lib/traject/macros/dor.rb

```diff
@@ -12,13 +12,55 @@ def stanford_mods(method, *args, **kwargs)
         end
       end
 
+      # Call a cocina_display method and accumulate the results
+      # @param method [Symbol] the cocina_display method to call
+      # @param kwargs [Hash] additional keyword arguments to pass to the cocina_display method
+      # @example
+      #   cocina_display(:pub_year_int, ignore_qualified: true)
+      def cocina_display(method, **kwargs)
+        lambda do |resource, accumulator, _context|
+          Array(resource.cocina_record.public_send(method, **kwargs)).each do |v|
+            accumulator << v
+          end
+        end
+      end
+
+      # Access Cocina record values at a JSON path expression and accumulate the results
+      # @param path [String] the JSON path expression to evaluate
+      # @example
+      #   cocina_display_path('$.description.note[?match(@.type, "table of contents")].value')
+      def cocina_display_path(path)
+        lambda do |resource, accumulator, _context|
+          Array(resource.cocina_record.path(path)).each do |v|
+            accumulator << v if v.present?
+          end
+        end
+      end
```

Comment:
These are pretty nice; I can see putting them into a traject-specific module in the Gem. When I get around to doing Earthworks/Searchworks indexing maybe it'll be time to do that